src/cloud/metrics/metrics_server.go

```diff
@@ -39,12 +42,33 @@ import (
 	"px.dev/pixie/src/shared/services/server"
 )
 
+var (
+	natsErrorCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "nats_error_count",
+		Help: "NATS message bus error",
+	}, []string{"shardID", "vizierID", "messageKind", "errorKind"})
+)
+
+// Extracts information from the subject and return shard, vizierID, messageType.
+func extractMessageInfo(subject string) (string, string, string) {
```

Comment:
What's the string expected to look like for the subscription? I believe the metrics server's subscription is created [here](https://github.com/pixie-io/pixie/blob/518183204f9f350ad42d064e77b39ada306e8d12/src/cloud/metrics/controllers/server.go#L124), but I'm not sure if that's what this function would receive.

Reply:
It will look something like `v2c.<shardID>.<vizierID>.VZMetrics`.
v2c stands for vizier to cloud, so a message originating from a Vizier to be sent to the cloud. 
`shardID` is the shard which the vizier sending the message belongs to, so that we can handle the subscriptions by shard. 
`vizierID` is the vizier of the ID sending the message.
`VZMetrics` is the channel which specifies the type of message.

We use this message format in our other services as well. for example vzmgr: https://github.com/pixie-io/pixie/blob/518183204f9f350ad42d064e77b39ada306e8d12/src/cloud/vzmgr/controllers/server.go#L118